builtin/providers/aws/resource_aws_redshift_cluster.go

```diff
@@ -345,24 +349,40 @@ func resourceAwsRedshiftClusterRead(d *schema.ResourceData, meta interface{}) er
 
 	d.Set("cluster_public_key", rsc.ClusterPublicKey)
 	d.Set("cluster_revision_number", rsc.ClusterRevisionNumber)
+	d.Set("tags", tagsToMapRedshift(rsc.Tags))
 
 	return nil
 }
 
 func resourceAwsRedshiftClusterUpdate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).redshiftconn
+	d.Partial(true)
 
+	arn, tagErr := buildRedshiftARN(d.Id(), meta.(*AWSClient).accountid, meta.(*AWSClient).region)
+	if tagErr != nil {
+		log.Printf("[DEBUG] Error building ARN for Redshift Cluster, not updating Tags for cluster %s", d.Id())
```

Comment:
I'm in favor of returning an error for a failed tag update, but would be happy to let it slide so long as there's a tracking ticket to come back to it.